blockchain/klaytn.go

```diff
@@ -0,0 +1,204 @@
+			Method:  "klay_blockNumber",
+		}
+
+		bytes, err := json.Marshal(msg)
+		if err != nil {
+			return nil
+		}
+
+		return bytes
+	}
+
+	return nil
+}
+
+// ParseResponse parses the response from the
+// Klaytn node, and returns a slice of subscriber.Events
+// and if the parsing was successful.
+//
+// If klaytnManager is using RPC:
+// If there are new events, update klaytnManager with
+// the latest block number it sees.
+func (k klaytnManager) ParseResponse(data []byte) ([]subscriber.Event, bool) {
+	logger.Debugw("Parsing response", "ExpectsMock", ExpectsMock)
+
+	var msg JsonrpcMessage
+	if err := json.Unmarshal(data, &msg); err != nil {
+		logger.Error("failed parsing msg: ", msg)
+		return nil, false
+	}
+
+	var events []subscriber.Event
+
+	switch k.p {
+	case subscriber.WS:
+		var res ethSubscribeResponse
+		if err := json.Unmarshal(msg.Params, &res); err != nil {
+			logger.Error("unmarshal:", err)
+			return nil, false
+		}
+
+		var evt models.Log
+		if err := json.Unmarshal(res.Result, &evt); err != nil {
+			logger.Error("unmarshal:", err)
+			return nil, false
+		}
+
+		request, err := logEventToOracleRequest(evt)
+		if err != nil {
+			logger.Error("failed to get oracle request:", err)
+			return nil, false
+		}
+
+		event, err := json.Marshal(request)
+		if err != nil {
+			logger.Error("marshal:", err)
+			return nil, false
+		}
+		logger.Warnw("receive message from subscribe", "evt", evt, "message", event)
+		events = append(events, event)
+
+	case subscriber.RPC:
+		var rawEvents []models.Log
+		if err := json.Unmarshal(msg.Result, &rawEvents); err != nil {
+			logger.Error("unmarshal:", err)
+			return nil, false
+		}
+
+		for _, evt := range rawEvents {
+			request, err := logEventToOracleRequest(evt)
+			if err != nil {
+				logger.Error("failed to get oracle request:", err, evt.Data, evt.Address)
+				return nil, false
+			}
+
+			event, err := json.Marshal(request)
+			if err != nil {
+				continue
+			}
+			events = append(events, event)
+
+			// Check if we can update the "fromBlock" in the query,
+			// so we only get new events from blocks we haven't queried yet
+			// Increment the block number by 1, since we want events from *after* this block number
+			curBlkn := &big.Int{}
+			curBlkn = curBlkn.Add(big.NewInt(int64(evt.BlockNumber)), big.NewInt(1))
+
+			fromBlkn, err := hexutil.DecodeBig(k.fq.FromBlock)
+			if err != nil && !(k.fq.FromBlock == "latest" || k.fq.FromBlock == "") {
+				logger.Error("Failed to get block number from event:", err)
+				continue
+			}
+
+			// If our query "fromBlock" is "latest", or our current "fromBlock" is in the past compared to
+			// the last event we received, we want to update the query
+			if k.fq.FromBlock == "latest" || k.fq.FromBlock == "" || curBlkn.Cmp(fromBlkn) > 0 {
+				k.fq.FromBlock = hexutil.EncodeBig(curBlkn)
+			}
+		}
+
+	default:
+		logger.Errorw(ErrSubscriberType.Error(), "type", k.p)
+		return nil, false
+	}
+
+	return events, true
+}
```

Comment:
Seems like you're missing the `blockchain/klaytn_test.go` file for testing the various functions?